File: programmers/js/basic/181912.js
// 배열 만들기 5
function solution(intStrs, k, s, l) {
  return intStrs.reduce((acc, cur) => {
    const num = +cur.slice(s, s + l);

    if (num > k) {
      acc.push(num);
    }

    return acc;
  }, []);
}

File: programmers/java/basic/181912.java
// 배열 만들기 5

import java.util.ArrayList;

class Solution {
    public int[] solution(String[] intStrs, int k, int s, int l) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (String intStr : intStrs) {
            int num = Integer.parseInt(intStr.substring(s, s + l));
            
            if (num > k) {
                result.add(num);
            }
        }
        
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}